privacyidea/lib/tokens/radiustoken.py

```diff
@@ -253,33 +508,47 @@ def check_otp(self, otpval, counter=None, window=None, options=None):
             if "transactionid" in options:
                 req["State"] = str(options.get("transactionid"))
 
+            if radius_state:
+                req["State"] = str(radius_state)
+                log.info("Sending saved challenge to radius server: {0} ".format(radius_state))
+
             response = srv.SendPacket(req)
-            c = response.code
-            # TODO: handle the RADIUS challenge
-            """
+
+            # handle the RADIUS challenge
             if response.code == pyrad.packet.AccessChallenge:
                 opt = {}
                 for attr in response.keys():
                     opt[attr] = response[attr]
                 res = False
-                log.debug("challenge returned %r " % opt)
+                log.info("Radiusserver challenge returned %s '%s' " % (opt["State"], opt["Reply-Message"]))
                 # now we map this to a privacyidea challenge
                 if "State" in opt:
-                    reply["transactionid"] = opt["State"][0]
+                    radius_state = opt["State"][0]
                 if "Reply-Message" in opt:
-                    reply["message"] = opt["Reply-Message"][0]
-            """
-            if response.code == pyrad.packet.AccessAccept:
+                    radius_message = opt["Reply-Message"][0]
+
+                result = 1
+            elif response.code == pyrad.packet.AccessAccept:
+                radius_state = '<SUCCESS>'
+                radius_message = 'RADIUS authentication succeeded'
                 log.info("Radiusserver %s granted "
                          "access to user %s." % (r_server, radius_user))
-                otp_count = 0
+                result = 0
             else:
-                log.warning("Radiusserver %s rejected "
-                            "access to user %s." % (r_server, radius_user))
+                radius_state = '<REJECTED>'
+                radius_message = 'RADIUS authentication failed'
+                log.debug('***response code %d' % response.code)
+                log.warning("Radiusserver %s "
```

Comment:
I would not want to call a warning in this case on the library level. This is a normal behaviour of a failed request.

Reply:
I have changed it to Info